packages/subgraph/src/helpers.ts

```diff
@@ -126,35 +125,32 @@ export function createOrUpdateNetworkEpochBlockNumber(
   return networkEpochBlockNumber;
 }
 
-export function getNetworkList(state: GlobalState): Array<Network> {
-  let result: Array<Network> = [];
-  if (state.networkArrayHead != null) {
-    let currentElement = Network.load(state.networkArrayHead!)!;
-    result.push(currentElement);
-    while (currentElement.nextArrayElement != null) {
-      currentElement = Network.load(currentElement.nextArrayElement!)!;
-      result.push(currentElement);
-      if (result.length > state.activeNetworkCount) {
-        log.warning(
-          "[getNetworkList] Network list processed is longer than activeNetworkCount. Network list length: {}, activeNetworkCount: {}",
-          [
-            result.length.toString(),
-            state.activeNetworkCount.toString(),
-          ]
-        );
-      }
+export function getActiveNetworks(state: GlobalState): Array<Network> {
+  let networks = new Array<Network>();
+  let nextId = state.networkArrayHead;
+
+  while (nextId != null) {
+    let network = Network.load(nextId!)!;
+    let isActive = network.removedAt == null;
+    if (isActive) {
+      networks.push(network);
     }
+    nextId = network.nextArrayElement;
   }
-  return result;
+
+  assert(
+    networks.length == state.activeNetworkCount,
+    `Found ${networks.length} active networks but ${state.activeNetworkCount} were expected. This is a bug!`,
+  );
```

Comment:
Adding `assert` here would break indexing when it triggers, if I recall correctly
This is of course beneficial to figure out that there's an issue, but we might not want to completely crash the subgraph if it happens